code/obj/item/bouquet.dm

```diff
@@ -0,0 +1,244 @@
+					src.inhand_image.overlays += image('icons/obj/items/bouquets.dmi', icon_state = "inhand_[flower.icon_state]_r")
+				else if (counter == 1)
+					src.overlays += image('icons/obj/items/bouquets.dmi', icon_state = "[flower.icon_state]_l")
+					src.inhand_image.overlays += image('icons/obj/items/bouquets.dmi', icon_state = "inhand_[flower.icon_state]_l")
+				else
+					src.overlays += image('icons/obj/items/bouquets.dmi', icon_state = "[flower.icon_state]_m")
+					src.inhand_image.overlays += image('icons/obj/items/bouquets.dmi', icon_state = "inhand_[flower.icon_state]_m")
+				counter += 1
+
+			src.overlays += image('icons/obj/items/bouquets.dmi', icon_state = "[src.wrapstyle]_front")
+
+			// all match
+			if ((flower1.icon_state == flower2.icon_state) && (flower2.icon_state == flower3.icon_state))
+				src.name = "[flower1.name] bouquet"
+				src.desc = "A bouquet of beautiful flowers. This one contains [flower1.name]."
+			// two match
+			else if ((flower1.icon_state == flower2.icon_state) || (flower2.icon_state == flower3.icon_state) || (flower1.icon_state == flower3.icon_state))
+				var/doubledflower = "" // the name of flower that matches
+				var/otherflower = "" // the name of the one that doesn't
+				if (flower1.icon_state == flower2.icon_state)
+					doubledflower = flower1.name
+					otherflower = flower3.name
+				else if (flower1.icon_state == flower3.icon_state)
+					doubledflower = flower1.name
+					otherflower = flower2.name
+				else if (flower2.icon_state == flower3.icon_state)
+					doubledflower = flower2.name
+					otherflower = flower1.name
+				src.name = "mixed bouquet"
+				src.desc = "A bouquet of beautiful flowers. This one contains [doubledflower] and [otherflower]."
+			// all different
+			else
+				src.name = "mixed bouquet"
+				src.desc = "A bouquet of beautiful flowers. This one contains [flower3.name], [flower2.name] and [flower1.name]."
+		if (4)
+			CRASH("Bouquet at [get_turf(src)] somehow has 4 flowers in it")
+	src.inhand_image.overlays += image('icons/obj/items/bouquets.dmi', icon_state = "inhand_[src.wrapstyle]_front")
+	if (src.hiddenitem)
+		src.desc += " There seems to be something else inside it as well."
+
+/// gently shakes the bouquet to indicate shuffling. mostly taken from hit_twitch()
+/obj/item/bouquet/proc/ruffle()
+	var/movepx = 0
+	var/movepy = 0
+	switch(pick(alldirs))
+		if (NORTH)
+			movepy = 3
+		if (WEST)
+			movepx = -3
+		if (SOUTH)
+			movepy = -3
+		if (EAST)
+			movepx = 3
+		if (NORTHEAST)
+			movepx = 2
+			movepy = 2
+		if (NORTHWEST)
+			movepx = -2
+			movepy = 2
+			movepy = -2
+		if (SOUTHEAST)
+			movepx = 2
+			movepy = -2
+		if (SOUTHWEST)
+			movepx = -2
+			movepy = -2
+		else
+			return
+	animate(src, pixel_x = movepx, pixel_y = movepy, time = 2, easing = EASE_IN, flags = ANIMATION_PARALLEL)
+	animate(pixel_x = movepx, pixel_y = movepy, time = 2, easing = EASE_IN)
```

Comment:
Aren't these the same parameters? What does this do?

Reply:
it does a little ruffling animation as far as i can tell. I'm totally unfamiliar with animation code so i copied this bit from somewhere else. It does work in game but there's probably a better way to do it.

Reply:
If you have a way to better implement an animation then i'm happy to do it.

Reply:
Animations tell it to go from the current state to the state specified. Both again calls are to the same thing. So the second one seems redundant.

Reply:
i see

Reply:
removed the second animation